x/staking/keeper/delegation.go

```diff
@@ -577,12 +577,12 @@ func (k Keeper) Unbond(
 
 	isValidatorOperator := delegation.DelegatorAddress.Equals(validator.OperatorAddress)
 
-	// if the delegation is the operator of the validator and undelegating will decrease the validator's self delegation below their minimum
-	// trigger a jail validator
+	// If the delegation is the operator of the validator and undelegating will decrease the validator's
+	// self-delegation below their minimum, we jail the validator.
 	if isValidatorOperator && !validator.Jailed &&
 		validator.TokensFromShares(delegation.Shares).TruncateInt().LT(validator.MinSelfDelegation) {
 
-		k.jailValidator(ctx, validator)
```

Comment:
Seems unnecessary to perform an extra store call to retrieve the validator object which we already have (as `k.Jail` uses the consensus address to retrieve the same validator object). But no big D

Reply:
Noted, but I switched for brevity. We've switched to Proto so the overhead is far lower now. I can switch back and add a comment 👍

Reply:
oh sweeet!